src/collector/top_collector.rs

```diff
@@ -86,32 +86,48 @@ where T: PartialOrd + Clone
 
     pub fn merge_fruits(
         &self,
-        children: Vec<Vec<(T, DocAddress)>>,
+        mut children: Vec<Vec<(T, DocAddress)>>,
     ) -> crate::Result<Vec<(T, DocAddress)>> {
         if self.limit == 0 {
             return Ok(Vec::new());
         }
-        let mut top_collector = BinaryHeap::new();
+        if children.len() == 1 {
+            let mut child = children.pop().unwrap();
+            child.sort_by(|a, b| {
+                if a.0 == b.0 {
+                    a.1.cmp(&b.1)
+                } else {
+                    b.0.partial_cmp(&a.0).unwrap_or(std::cmp::Ordering::Equal)
+                }
+            });
+            if self.offset != 0 {
+                return Ok(child
+                    .into_iter()
+                    .skip(self.offset)
+                    .take(self.limit)
+                    .collect());
+            }
+            if child.len() > self.limit {
```

Comment:
I think `Vec::truncate` already does this check internally, so the call could be unconditional.